packages/terra-table/src/SelectableTableRows.jsx

```diff
@@ -31,24 +35,77 @@ const propTypes = {
 const defaultProps = {
   disableUnselectedRows: false,
   onChange: undefined,
+  onBlur: undefined,
   selectedIndexes: [],
 };
 
+const rowsSelector = 'row-group';
+
 class SelectableTableRows extends React.Component {
   constructor(props) {
     super(props);
     this.handleOnChange = this.handleOnChange.bind(this);
+    this.handleOnBlur = this.handleOnBlur.bind(this);
+  }
+
+
+  componentDidMount() {
+    // Adds ARIA-Label for Pre-selected rows
+    const rowGroup = document.querySelector(`#${rowsSelector}`);
+    for (let rowIndex = 0; rowIndex < rowGroup.children.length; rowIndex += 1) {
+      const preselectedRow = rowGroup.children[rowIndex];
+      let ariaLabel = 'pre-selected row';
+      if ((preselectedRow.hasAttribute('aria-selected')) && (preselectedRow.getAttributeNode('aria-selected').value === 'true')) {
+        for (let columnIndex = 0; columnIndex < preselectedRow.children.length; columnIndex += 1) {
+          ariaLabel += ` ${preselectedRow.children[columnIndex].innerText}`;
+        }
+        rowGroup.children[rowIndex].setAttribute('aria-label', ariaLabel);
+      }
+    }
   }
 
   handleOnChange(event, index) {
     if (this.props.onChange) {
       this.props.onChange(event, index);
     }
+
+    if (!document.activeElement.hasAttribute('aria-label')) {
```

Comment:
What happens if I select a row that already has some other value set to `aria-label`?

Reply:
this check is added to prevent adding aria-label for pre-selected rows  and for the row which has already got aria-label with it. This will happen when click on same row again. 

Since there is no default aria-label is added to rows initially other than for pre-selected rows, row will not be having aria-label with some other value as of now. 

Aria-Label for preselected-rows will not be overridden since pre-selected rows cannot be selected again on click or on key down.